packages/edit-post/src/components/header/index.js

```diff
@@ -40,13 +41,12 @@ function Header( { setEntitiesSavedStatesCallback } ) {
 			showIconLabels:
 				select( editPostStore ).isFeatureActive( 'showIconLabels' ),
 			isDistractionFree:
-				select( editPostStore ).isFeatureActive( 'distractionFree' ),
+				select( editPostStore ).isFeatureActive( 'distractionFree' ) &&
+				isLargeViewport,
 		} ),
 		[]
```

Comment:
```suggestion
		[ isLargeViewport ]
```

You'll want to re-run the selector if the viewport changes right?

Reply:
This seems still unfixed? Can we commit that?

Reply:
I raised https://github.com/WordPress/gutenberg/pull/47611